File: android/app/src/androidTest/java/my/ssdid/wallet/ui/offline/InMemoryBundleStore.kt
package my.ssdid.wallet.ui.offline

import my.ssdid.sdk.domain.verifier.offline.BundleStore
import my.ssdid.sdk.domain.verifier.offline.VerificationBundle

class InMemoryBundleStore : BundleStore {
    private val bundles = mutableMapOf<String, VerificationBundle>()

    override suspend fun saveBundle(bundle: VerificationBundle) {
        bundles[bundle.issuerDid] = bundle
    }

    override suspend fun getBundle(issuerDid: String): VerificationBundle? {
        return bundles[issuerDid]
    }

    override suspend fun deleteBundle(issuerDid: String) {
        bundles.remove(issuerDid)
    }

    override suspend fun listBundles(): List<VerificationBundle> {
        return bundles.values.toList()
    }

    fun clear() { bundles.clear() }
}

File: ios/SsdidWalletTests/Offline/InMemoryBundleStore.swift
@testable import SsdidCore
import Foundation
@testable import SsdidWallet

final class InMemoryBundleStore: BundleStore, @unchecked Sendable {
    private var bundles: [String: VerificationBundle] = [:]

    func saveBundle(_ bundle: VerificationBundle) async throws {
        bundles[bundle.issuerDid] = bundle
    }

    func getBundle(issuerDid: String) async -> VerificationBundle? {
        bundles[issuerDid]
    }

    func deleteBundle(issuerDid: String) async throws {
        bundles.removeValue(forKey: issuerDid)
    }

    func listBundles() async throws -> [VerificationBundle] {
        Array(bundles.values)
    }

    func clear() { bundles.removeAll() }
}
